src/util/rpc/http/kqueue_event_handler.cpp

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2022 MIT Digital Currency Initiative,
+//                    Federal Reserve Bank of Boston
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "kqueue_event_handler.hpp"
+
+#include <algorithm>
+#include <array>
+#include <iostream>
+#include <unistd.h>
+
+namespace cbdc::rpc {
+    kqueue_event_handler::~kqueue_event_handler() {
+        close(m_kq);
+    }
+
+    auto kqueue_event_handler::init() -> bool {
+        if((m_kq = kqueue()) == -1) {
+            return false;
+        }
+        return true;
```

Comment:
```suggestion
        m_kq = kqueue();
        return m_kq != -1;
```

Style nit; no reason to prefer inline-assignment inside the `if`, especially when the `if` is of the always-redundant form `if (condition) { return bool; } else { return bool; }`

Reply:
Agreed on separating `m_kq = kqueue();`. However, I suggest we keep the if/else structure for clarity. At least for me reading `return m_kq != -1`, despite its brevity, requires some brain cycles.

Reply:
(Sorry if this seemed silly; I realize it probably did. It's just one of those ones that bothers me for some reason. Take nothing personal about it, please!)

Reply:
It bothers me, but as it is marked as a nit, you can safely read it as "I'm not particularly moved by this one; take it or leave it"; so, if you don't find this compelling, please do feel close it and I won't be mad in any fashion.